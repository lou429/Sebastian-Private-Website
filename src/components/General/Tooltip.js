import React from 'react'; 
import './Tooltip.scss';

function Tooltip(props) {
    return(
        <div className="tooltip">
            {props.children}
            <span className="toolTipText">{props.text}</span>
        </div>
    );
}

export default Tooltip;