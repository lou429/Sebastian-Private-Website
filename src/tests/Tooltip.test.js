import {render} from '@testing-library/react';
import Tooltip from '../components/tooltip';

test('Tooltip snapshot test', () => {
    const { container } = render(
        <Tooltip text="Test render">
            <h1>Heading</h1>
        </Tooltip>
    );

    expect(container).toMatchSnapshot();
})

test('Can tooltips render', () => {
    let text = "Tooltip test";

    const { getByText } = render(
        <Tooltip text={text}>
            <h1>Heading</h1>
        </Tooltip>
    );

    expect(getByText(text)).toBeInTheDocument();
});
