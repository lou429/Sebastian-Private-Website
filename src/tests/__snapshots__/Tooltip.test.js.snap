// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip snapshot test 1`] = `
<div>
  <div
    class="tooltip"
  >
    <h1>
      Heading
    </h1>
    <span
      class="toolTipText"
    >
      Test render
    </span>
  </div>
</div>
`;
